manipulator_controller:
  ros__parameters:
    protocol: "usb"
    device_id: 1701538158

    command_publish_rate: 100.0
    feedback_rate: 100.0
    motor_control_rate: 1000.0

    joint_trajectory_topic: "/manipulator/joint_trajectory"
    joint_states_topic: "/manipulator/joint_states"

    manipulator_joint_names: ["joint_1", "joint_2", "joint_3", "gripper_1", "gripper_2", "gripper_3"]

    max_pwm_dutycycle: 80.0
    velocity_filter_cutoff: 100.0

    pid_kp: 0.0
    pid_ki: 0.0
    pid_kd: 0.0

    pid_p_bound: 0.0
    pid_i_bound: 0.0
    pid_d_bound: 0.0

    ff_kv: 1.0
    ff_ka: 0.0
    ff_kj: 0.0

    overwrite_pinout: true
    joint_dir_pins: [3, 7, 9]
    joint_pul_pins: [2, 6, 8]
    gripper_lpwm_pins: [14, 10, 12]
    gripper_rpwm_pins: [15, 11, 13]

    joint_swap_dirs: [0, 0, 0]
    gripper_swap_dirs: [0, 0, 0]

    joint_initial_positions: [0.0, 0.0, 0.0]
    joint_min_pos_boundaries: [-3.1416, -3.1416, -3.1416]
    joint_max_pos_boundaries: [3.1416, 3.1416, 3.1416]

    joint_reductions: [50.0, 68.18181818, 2.0]
    joint_steps_per_revolutions: [20000.0, 12800.0, 51200.0]

    base_frame_id: base_link

    joint_trajectory_timeout: 0.5

    linear.x.has_velocity_limits: true
    #linear.x.has_acceleration_limits: true
    # linear.x.has_jerk_limits: false
    linear.x.max_velocity: 2.0
    linear.x.min_velocity: -2.0
    #linear.x.max_acceleration: 5.0
    #linear.x.min_acceleration: -5.0
    # linear.x.max_jerk: NAN
    # linear.x.min_jerk: NAN

    angular.z.has_velocity_limits: true
    #angular.z.has_acceleration_limits: true
    # angular.z.has_jerk_limits: false
    angular.z.max_velocity: 2.0
    angular.z.min_velocity: -2.0
    #angular.z.max_acceleration: 5.0
    #angular.z.min_acceleration: -5.0
    # angular.z.max_jerk: NAN
    # angular.z.min_jerk: NAN
